pinot-core/src/main/java/com/linkedin/pinot/core/util/IdMap.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.util;
+
+/**
+ * Interface for a map from Key to auto-generated contiguous integer Id's.
+ *
+ * @param <Key>
+ */
+public interface IdMap<Key> {
+  int INVALID_ID = -1;
+
+  /**
+   * Puts the specified key into the map. The id for the key is auto-generated as follows:
+   *
+   * <ul> If key does not exist in the map, it is inserted into the map with a new id which is equal to number of
+   * elements in the map before the key is inserted. For example, the first key inserted will have id of 0, second
+   * will have an id of 1, and so on. </ul>
+   *
+   * <ul> If key already exists in the map, then it keeps its original id.</ul>
+   *
+   * @param key Key to be inserted into the map.
+   */
+  void put(Key key);
```

Comment:
any reason why we are not returning the id

Reply:
Yes, I thought about that too. The current off-heap dictionary based implementation does not return the id, so it will have to do an explict get internally to return the id. Nonetheless, I think it is a good idea to change the interface to return the id. Will update.